foster55f/market2table-fe
# Phase 1 FE Implementation

### What does this PR do?
 - This PR implements the FE functionality for phase 1
 - Creates the initial file structure for the front end repo
 - Implements Redux using actions/reducers to create the store and using useSelector and useDispatch to connect components to store - HOOKS.
- Implements React Hooks on all components so far
- Implements Router to create a multiple page app and fluid navigation
- Has basic functionality of inputting a zip code and getting the top 20 markets based off of that zip code - the markets are listed and displayed on google maps using google-maps-react library connected with Google Maps API - Each market can be selected to display that market's information and isolate it on the map. The zip code can be updated without going back to the main screen.
- Creates an environmental variable for the backend URL as well as the Google Maps API key so that it is not accessible via GH
 - Implements error handling for the zipCodeForm to only accept 6 digits

### How to test?
 - Pull down branch
 - run 'npm start' and the site should open in browser via localhost:3000
 - Enter a zip code and hit Find!
 - Navigate through site

### Issues:
 - We are currently using a nested GET request to the Farmers Market API - once our backend is up and running then we will need to get rid of that and update to perform a GET to our deployed backend. This will require changing and updating the display and format of some things, but not much
 - We need to fine-tune the styling for the header and the marketsContainer/Market/MarketPage components and how that information is being displayed.

### Notes:
- We should wireframe the rest of the app out before proceeding any further with FE functionality implementation. Figure out what exactly needs to be in the store - how we will need to use the backend - user flow and functionality.


### Relevant Screenshots:
<img width="1440" alt="Screen Shot 2020-02-17 at 10 07 54 PM" src="https://user-images.githubusercontent.com/49846853/74705888-9e5b7f80-51d2-11ea-9ccd-b86e3d783d4b.png">

<img width="1440" alt="Screen Shot 2020-02-17 at 10 08 22 PM" src="https://user-images.githubusercontent.com/49846853/74705922-ba5f2100-51d2-11ea-9735-2ea2bf65ca4a.png">

<img width="1440" alt="Screen Shot 2020-02-17 at 10 08 28 PM" src="https://user-images.githubusercontent.com/49846853/74705931-bdf2a800-51d2-11ea-8f1d-c44fa16e27fc.png">

Thanks!
@foster55f
